src/runtime-rs/crates/resource/src/volume/hugepage.rs

```diff
@@ -0,0 +1,224 @@
+
+        // Create a storage struct so that kata agent is able to create
+        // hugetlbfs backed volume inside the VM
+        let storage = Storage {
+            driver: KATA_EPHEMERAL_VOLUME_TYPE.to_string(),
+            source: NODEV.to_string(),
+            fs_type: HUGETLBFS.to_string(),
+            mount_point: mount.source.clone(),
+            options: vec![option],
+            ..Default::default()
+        };
+        Ok(Self {
+            storage: Some(storage),
+            mount,
+        })
+    }
+}
+
+impl Volume for Hugepage {
+    fn get_volume_mount(&self) -> Result<Vec<oci::Mount>> {
+        Ok(vec![self.mount.clone()])
+    }
+
+    fn get_storage(&self) -> Result<Vec<agent::Storage>> {
+        let s = if let Some(s) = self.storage.as_ref() {
+            vec![s.clone()]
+        } else {
+            vec![]
+        };
+        Ok(s)
+    }
+
+    fn cleanup(&self) -> Result<()> {
+        todo!()
+    }
+}
+
+pub(crate) fn is_huge_page(m: &oci::Mount) -> Result<(bool, Option<Vec<String>>)> {
+    if m.source.is_empty() {
+        return Err(anyhow!("empty mount source"));
+    }
+    let file = File::open(PROC_MOUNTS_FILE).context("failed open file")?;
+    let reader = BufReader::new(file);
+    for line in reader.lines().flatten() {
+        let items: Vec<&str> = line.split(' ').collect();
+        if m.source == items[1] && items[2] == HUGETLBFS {
+            let fs_options: Vec<&str> = items[3].split(',').collect();
+            return Ok((
+                true,
+                Some(
+                    fs_options
+                        .iter()
+                        .map(|&s| s.to_string())
+                        .collect::<Vec<String>>(),
+                ),
+            ));
+        }
+    }
+    Ok((false, None))
+}
+
+pub(crate) fn get_huge_page_option(spec: &oci::Spec) -> Result<Option<HashMap<Byte, u64>>> {
+    if let Some(l) = &spec.linux {
+        if let Some(r) = &l.resources {
+            let hugepage_limits = r.hugepage_limits.clone();
+            let mut options_map: HashMap<Byte, u64> = HashMap::new();
+            for hugepage_limit in hugepage_limits {
+                // the pagesize send from oci spec is MB or GB, change it to Mi and Gi
+                let page_size = hugepage_limit.page_size.replace('B', "i");
+                let page_size = Byte::from_str(page_size)
+                    .context("failed to create Byte object from String")?;
+                options_map.insert(page_size, hugepage_limit.limit);
+            }
+            return Ok(Some(options_map));
+        }
+        return Ok(None);
+    }
+    Ok(None)
+}
+
+fn get_page_size(fs_options: Option<Vec<String>>) -> Option<String> {
+    if let Some(fs_options) = fs_options {
+        for fs_option in fs_options {
+            if fs_option.starts_with("pagesize=") {
+                return fs_option
+                    .strip_prefix("pagesize=")
+                    .map(|s| format!("{}i", s));
```

Comment:
Why format with `"{}i"`, it is recommended to add a comment.

Reply:
fixed